### ISSUE TITLE
Crash when changing split file names

### ISSUE THREAD

**Issue by Herohtar** [josemesona/ManagedWimgApi#27]
I'm capturing an image and then splitting the resulting WIM file like this:

```C#
using (WimHandle wimHandle = WimgApi.CreateFile(fullImageFile, WimFileAccess.Write, WimCreationDisposition.CreateAlways, WimCreateFileOptions.None, WimCompressionType.Lzx))
{
    WimgApi.SetTemporaryPath(wimHandle, Environment.GetEnvironmentVariable("systemdrive"));
    WimgApi.RegisterMessageCallback(wimHandle, CaptureCallback);

    try
    {
        using (WimHandle imageHandle = WimgApi.CaptureImage(wimHandle, captureFolder, WimCaptureImageOptions.None))
        {
            WimgApi.SplitFile(wimHandle, splitImageFile, SplitFileSize);
        }
    }
    catch (Exception e)
    {
        // Exception handling stuff
    }
    finally
    {
        WimgApi.UnregisterMessageCallback(wimHandle, CaptureCallback);
    }
}
```

It works fine if that's all I do; however, if I try to change the path of the split file parts in my `CaptureCallback` the program crashes, hard. It doesn't throw any exceptions (catch/finally is never reached) and no error messages. The entire program terminates. My capture is running in its own thread, so typically if there is an uncaught error it just results in the thread dying, but whatever is going on here brings everything down and not even the debugger can see it.

The problem code looks like this:
```
case WimMessageType.Split:
    WimMessageSplit splitMessage = (WimMessageSplit)message;
    splitMessage.PartPath = newFileName;
    break;
```

I can make the crash go away simply by commenting out the one line:
```
case WimMessageType.Split:
    WimMessageSplit splitMessage = (WimMessageSplit)message;
    //splitMessage.PartPath = newFileName; // No more crash
    break;
```

One more thing, it seems that the crash happens at the end of the `.SplitFile` call -- that is, all the parts get written, with the correct (modified) name, but at the point where the call should be returning to my code, something breaks.

**Comment by Herohtar** [comment]
I've submitted PR #28 which I believe fixes the issue. I think the problem is that when you set `PartPath`, the pointer in `wParam` is originally pointing to a string pointer that was allocated by the native API, so when you call `Marshal.FreeHGlobal(Marshal.ReadIntPtr(wParam))` it's freeing memory that should be managed by the API. My guess is that the API then later tries to access the original string in some way, possibly to free it itself, but since you've already freed the memory it crashes depending on whether or not that memory got taken up by something else in the meantime.

**Comment by jeffkl** [comment]
I have published version 1.2.4 of the package with this fix.
